src/raftstore/store/fsm/peer.rs

```diff
@@ -1862,19 +1862,20 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
             );
         }
         let target = state.get_target().get_id();
+        let commit = state.get_commit();
         self.fsm.peer.pending_merge_state = Some(state);
         self.notify_prepare_merge();
 
         if let Some(logs_up_to_date) = self.fsm.peer.catch_up_logs.take() {
-            logs_up_to_date.store(region.get_id(), Ordering::SeqCst);
-            let mailbox = self.ctx.apply_router.mailbox(target).unwrap();
             // Send CatchUpLogs back to destroy source apply delegate,
             // then it will send `LogsUpToDate` to target apply delegate.
+            let mut req = CommitMergeRequest::new();
+            req.set_commit(commit);
```

Comment:
Is it necessary to attach commit?

Reply:
Not necessary, but it's better to take it to double-check `apply_index >= commit`